File: api/src/adapters/config-manager.ts
/**
 * LLM-Config-Manager Adapter
 *
 * Thin adapter for consuming configuration-driven analytics parameters
 * and retention settings from LLM-Config-Manager.
 */

import { logger } from '../logger';
import {
  AdapterHealth,
  EcosystemAdapter,
  healthyAdapter,
  unhealthyAdapter,
} from './types';

export interface ConfigManagerConfig {
  endpoint: string;
  apiKey?: string;
  timeoutMs?: number;
  cacheTtlSecs?: number;
}

/**
 * Analytics parameters
 */
export interface AnalyticsParameters {
  configId: string;
  version: string;
  createdAt: Date;
  aggregation: AggregationConfig;
  anomalyDetection: AnomalyDetectionConfig;
  forecasting: ForecastingConfig;
  alerting: AlertingConfig;
  sampling: SamplingConfig;
}

export interface AggregationConfig {
  defaultWindowMinutes: number;
  rollupWindows: RollupWindow[];
  defaultPercentiles: number[];
  maxCardinality: number;
  enableHistograms: boolean;
}

export interface RollupWindow {
  name: string;
  durationMinutes: number;
  aggregations: string[];
}

export interface AnomalyDetectionConfig {
  enabled: boolean;
  algorithm: 'z_score' | 'iqr' | 'dbscan' | 'isolation_forest' | 'prophet';
  sensitivity: number;
  minDataPoints: number;
  evaluationWindowMinutes: number;
  cooldownMinutes: number;
}

export interface ForecastingConfig {
  enabled: boolean;
  model: 'arima' | 'prophet' | 'exponential_smoothing' | 'linear_regression' | 'lstm';
  horizonHours: number;
  trainingWindowDays: number;
  updateFrequencyHours: number;
  confidenceLevel: number;
}

export interface AlertingConfig {
  enabled: boolean;
  defaultSeverity: 'info' | 'warning' | 'error' | 'critical';
  channels: AlertChannel[];
  rateLimitPerHour: number;
  groupingWindowMinutes: number;
}

export interface AlertChannel {
  channelType: 'email' | 'slack' | 'pagerduty' | 'webhook' | 'sns';
  config: Record<string, string>;
  enabled: boolean;
}

export interface SamplingConfig {
  enabled: boolean;
  defaultRate: number;
  highVolumeRate: number;
  highVolumeThresholdRps: number;
  preserveErrors: boolean;
}

/**
 * Retention settings
 */
export interface RetentionSettings {
  configId: string;
  version: string;
  createdAt: Date;
  policies: RetentionPolicy[];
  archival: ArchivalConfig;
  compaction: CompactionConfig;
}

export interface RetentionPolicy {
  policyId: string;
  name: string;
  dataType: 'raw_events' | 'aggregated_metrics' | 'traces' | 'logs' | 'alerts' | 'audits';
  retentionDays: number;
  tier: 'hot' | 'warm' | 'cold' | 'archive';
  compressAfterDays?: number;
  archiveAfterDays?: number;
}

export interface ArchivalConfig {
  enabled: boolean;
  destination: ArchivalDestination;
  compression: 'none' | 'gzip' | 'zstd' | 'lz4' | 'snappy';
  encryptionEnabled: boolean;
}

export type ArchivalDestination =
  | { type: 's3'; bucket: string; prefix: string }
  | { type: 'gcs'; bucket: string; prefix: string }
  | { type: 'azure'; container: string; prefix: string };

export interface CompactionConfig {
  enabled: boolean;
  scheduleCron: string;
  targetFileSizeMb: number;
  maxConcurrentJobs: number;
}

/**
 * Feature flags
 */
export interface FeatureFlags {
  configId: string;
  flags: Record<string, FeatureFlag>;
  lastUpdated: Date;
}

export interface FeatureFlag {
  name: string;
  enabled: boolean;
  description: string;
  rolloutPercentage: number;
  allowedEnvironments: string[];
}

/**
 * Environment config
 */
export interface EnvironmentConfig {
  environment: string;
  endpoints: Record<string, string>;
  limits: ResourceLimits;
  security: SecurityConfig;
}

export interface ResourceLimits {
  maxConcurrentQueries: number;
  maxQueryTimeoutSecs: number;
  maxResultRows: number;
  maxMemoryMb: number;
}

export interface SecurityConfig {
  requireAuth: boolean;
  allowedOrigins: string[];
  rateLimitRps: number;
  ipWhitelist?: string[];
}

/**
 * LLM-Config-Manager adapter for consuming configuration data
 */
export class ConfigManagerAdapter implements EcosystemAdapter {
  private config: ConfigManagerConfig;
  private connected: boolean = false;

  constructor(config?: Partial<ConfigManagerConfig>) {
    this.config = {
      endpoint: process.env.CONFIG_MANAGER_ENDPOINT || 'http://localhost:8085',
      apiKey: process.env.CONFIG_MANAGER_API_KEY,
      timeoutMs: parseInt(process.env.CONFIG_MANAGER_TIMEOUT_MS || '30000', 10),
      cacheTtlSecs: parseInt(process.env.CONFIG_MANAGER_CACHE_TTL_SECS || '300', 10),
      ...config,
    };
  }

  async connect(): Promise<void> {
    logger.info({ endpoint: this.config.endpoint }, 'Connecting to LLM-Config-Manager');
    this.connected = true;
    logger.info('Successfully connected to LLM-Config-Manager');
  }

  async healthCheck(): Promise<AdapterHealth> {
    const start = Date.now();

    if (!this.connected) {
      return unhealthyAdapter('config_manager', 'Not connected');
    }

    const latencyMs = Date.now() - start;
    return healthyAdapter('config_manager', latencyMs);
  }

  async disconnect(): Promise<void> {
    logger.info('Disconnecting from LLM-Config-Manager');
    this.connected = false;
  }

  /**
   * Fetch analytics parameters
   */
  async fetchAnalyticsParameters(): Promise<AnalyticsParameters> {
    if (!this.connected) {
      throw new Error('Config-Manager adapter not connected');
    }

    logger.debug('Fetching analytics parameters from Config-Manager');

    // Placeholder with sensible defaults
    return {
      configId: crypto.randomUUID(),
      version: '1.0.0',
      createdAt: new Date(),
      aggregation: {
        defaultWindowMinutes: 5,
        rollupWindows: [
          { name: '1min', durationMinutes: 1, aggregations: ['avg', 'count'] },
          { name: '5min', durationMinutes: 5, aggregations: ['avg', 'min', 'max', 'count'] },
          { name: '1hour', durationMinutes: 60, aggregations: ['avg', 'min', 'max', 'p50', 'p95', 'p99', 'count'] },
        ],
        defaultPercentiles: [0.5, 0.9, 0.95, 0.99],
        maxCardinality: 10000,
        enableHistograms: true,
      },
      anomalyDetection: {
        enabled: true,
        algorithm: 'z_score',
        sensitivity: 3.0,
        minDataPoints: 30,
        evaluationWindowMinutes: 15,
        cooldownMinutes: 60,
      },
      forecasting: {
        enabled: true,
        model: 'exponential_smoothing',
        horizonHours: 24,
        trainingWindowDays: 7,
        updateFrequencyHours: 1,
        confidenceLevel: 0.95,
      },
      alerting: {
        enabled: true,
        defaultSeverity: 'warning',
        channels: [],
        rateLimitPerHour: 100,
        groupingWindowMinutes: 5,
      },
      sampling: {
        enabled: false,
        defaultRate: 1.0,
        highVolumeRate: 0.1,
        highVolumeThresholdRps: 10000,
        preserveErrors: true,
      },
    };
  }

  /**
   * Fetch retention settings
   */
  async fetchRetentionSettings(): Promise<RetentionSettings> {
    if (!this.connected) {
      throw new Error('Config-Manager adapter not connected');
    }

    logger.debug('Fetching retention settings from Config-Manager');

    // Placeholder with sensible defaults
    return {
      configId: crypto.randomUUID(),
      version: '1.0.0',
      createdAt: new Date(),
      policies: [
        {
          policyId: 'raw-events',
          name: 'Raw Events',
          dataType: 'raw_events',
          retentionDays: 7,
          tier: 'hot',
          compressAfterDays: 1,
          archiveAfterDays: 7,
        },
        {
          policyId: 'aggregated-metrics',
          name: 'Aggregated Metrics',
          dataType: 'aggregated_metrics',
          retentionDays: 90,
          tier: 'warm',
          compressAfterDays: 7,
          archiveAfterDays: 30,
        },
        {
          policyId: 'traces',
          name: 'Traces',
          dataType: 'traces',
          retentionDays: 14,
          tier: 'hot',
          compressAfterDays: 3,
          archiveAfterDays: 14,
        },
      ],
      archival: {
        enabled: false,
        destination: { type: 's3', bucket: 'analytics-archive', prefix: 'data/' },
        compression: 'zstd',
        encryptionEnabled: true,
      },
      compaction: {
        enabled: true,
        scheduleCron: '0 2 * * *',
        targetFileSizeMb: 256,
        maxConcurrentJobs: 4,
      },
    };
  }

  /**
   * Fetch feature flags
   */
  async fetchFeatureFlags(): Promise<FeatureFlags> {
    if (!this.connected) {
      throw new Error('Config-Manager adapter not connected');
    }

    logger.debug('Fetching feature flags from Config-Manager');

    // Placeholder implementation
    return {
      configId: crypto.randomUUID(),
      flags: {},
      lastUpdated: new Date(),
    };
  }

  /**
   * Fetch environment config
   */
  async fetchEnvironmentConfig(environment: string): Promise<EnvironmentConfig> {
    if (!this.connected) {
      throw new Error('Config-Manager adapter not connected');
    }

    logger.debug({ environment }, 'Fetching environment config from Config-Manager');

    // Placeholder implementation
    return {
      environment,
      endpoints: {},
      limits: {
        maxConcurrentQueries: 100,
        maxQueryTimeoutSecs: 300,
        maxResultRows: 100000,
        maxMemoryMb: 4096,
      },
      security: {
        requireAuth: true,
        allowedOrigins: ['*'],
        rateLimitRps: 1000,
      },
    };
  }

  /**
   * Get config value by key
   */
  async getConfigValue<T>(key: string): Promise<T | null> {
    if (!this.connected) {
      throw new Error('Config-Manager adapter not connected');
    }

    logger.debug({ key }, 'Fetching config value from Config-Manager');

    // Placeholder implementation
    return null;
  }
}

File: src/adapters/config_manager.rs
//! LLM-Config-Manager Adapter
//!
//! Thin adapter for consuming configuration-driven analytics parameters
//! and retention settings from LLM-Config-Manager.
//!
//! This adapter provides read-only access to configuration data for analytics
//! purposes without modifying any upstream logic.

use super::{AdapterHealth, EcosystemAdapter};
use anyhow::Result;
use async_trait::async_trait;
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use std::sync::atomic::{AtomicBool, Ordering};
use std::time::Instant;
use tracing::{debug, info, instrument};

/// Configuration for Config-Manager adapter
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConfigManagerConfig {
    pub endpoint: String,
    pub api_key: Option<String>,
    pub timeout_secs: u64,
    pub cache_ttl_secs: u64,
}

impl ConfigManagerConfig {
    pub fn from_env() -> Result<Self> {
        Ok(Self {
            endpoint: std::env::var("CONFIG_MANAGER_ENDPOINT")
                .unwrap_or_else(|_| "http://localhost:8085".to_string()),
            api_key: std::env::var("CONFIG_MANAGER_API_KEY").ok(),
            timeout_secs: std::env::var("CONFIG_MANAGER_TIMEOUT_SECS")
                .unwrap_or_else(|_| "30".to_string())
                .parse()
                .unwrap_or(30),
            cache_ttl_secs: std::env::var("CONFIG_MANAGER_CACHE_TTL_SECS")
                .unwrap_or_else(|_| "300".to_string())
                .parse()
                .unwrap_or(300),
        })
    }
}

/// Analytics parameters configuration
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AnalyticsParameters {
    pub config_id: String,
    pub version: String,
    pub created_at: DateTime<Utc>,
    pub aggregation: AggregationConfig,
    pub anomaly_detection: AnomalyDetectionConfig,
    pub forecasting: ForecastingConfig,
    pub alerting: AlertingConfig,
    pub sampling: SamplingConfig,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AggregationConfig {
    pub default_window_minutes: u32,
    pub rollup_windows: Vec<RollupWindow>,
    pub default_percentiles: Vec<f64>,
    pub max_cardinality: u64,
    pub enable_histograms: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RollupWindow {
    pub name: String,
    pub duration_minutes: u32,
    pub aggregations: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AnomalyDetectionConfig {
    pub enabled: bool,
    pub algorithm: AnomalyAlgorithm,
    pub sensitivity: f64,
    pub min_data_points: u32,
    pub evaluation_window_minutes: u32,
    pub cooldown_minutes: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum AnomalyAlgorithm {
    ZScore,
    IQR,
    DBSCAN,
    IsolationForest,
    Prophet,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ForecastingConfig {
    pub enabled: bool,
    pub model: ForecastModel,
    pub horizon_hours: u32,
    pub training_window_days: u32,
    pub update_frequency_hours: u32,
    pub confidence_level: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ForecastModel {
    ARIMA,
    Prophet,
    ExponentialSmoothing,
    LinearRegression,
    LSTM,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AlertingConfig {
    pub enabled: bool,
    pub default_severity: AlertSeverity,
    pub channels: Vec<AlertChannel>,
    pub rate_limit_per_hour: u32,
    pub grouping_window_minutes: u32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum AlertSeverity {
    Info,
    Warning,
    Error,
    Critical,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AlertChannel {
    pub channel_type: ChannelType,
    pub config: HashMap<String, String>,
    pub enabled: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ChannelType {
    Email,
    Slack,
    PagerDuty,
    Webhook,
    SNS,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SamplingConfig {
    pub enabled: bool,
    pub default_rate: f64,
    pub high_volume_rate: f64,
    pub high_volume_threshold_rps: u64,
    pub preserve_errors: bool,
}

/// Data retention settings
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RetentionSettings {
    pub config_id: String,
    pub version: String,
    pub created_at: DateTime<Utc>,
    pub policies: Vec<RetentionPolicy>,
    pub archival: ArchivalConfig,
    pub compaction: CompactionConfig,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RetentionPolicy {
    pub policy_id: String,
    pub name: String,
    pub data_type: DataType,
    pub retention_days: u32,
    pub tier: StorageTier,
    pub compress_after_days: Option<u32>,
    pub archive_after_days: Option<u32>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum DataType {
    RawEvents,
    AggregatedMetrics,
    Traces,
    Logs,
    Alerts,
    Audits,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum StorageTier {
    Hot,
    Warm,
    Cold,
    Archive,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ArchivalConfig {
    pub enabled: bool,
    pub destination: ArchivalDestination,
    pub compression: CompressionType,
    pub encryption_enabled: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ArchivalDestination {
    S3 { bucket: String, prefix: String },
    GCS { bucket: String, prefix: String },
    Azure { container: String, prefix: String },
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum CompressionType {
    None,
    Gzip,
    Zstd,
    Lz4,
    Snappy,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CompactionConfig {
    pub enabled: bool,
    pub schedule_cron: String,
    pub target_file_size_mb: u64,
    pub max_concurrent_jobs: u32,
}

/// Feature flags for analytics
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeatureFlags {
    pub config_id: String,
    pub flags: HashMap<String, FeatureFlag>,
    pub last_updated: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeatureFlag {
    pub name: String,
    pub enabled: bool,
    pub description: String,
    pub rollout_percentage: f64,
    pub allowed_environments: Vec<String>,
}

/// Environment-specific configuration
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EnvironmentConfig {
    pub environment: String,
    pub endpoints: HashMap<String, String>,
    pub limits: ResourceLimits,
    pub security: SecurityConfig,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ResourceLimits {
    pub max_concurrent_queries: u32,
    pub max_query_timeout_secs: u32,
    pub max_result_rows: u64,
    pub max_memory_mb: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SecurityConfig {
    pub require_auth: bool,
    pub allowed_origins: Vec<String>,
    pub rate_limit_rps: u32,
    pub ip_whitelist: Option<Vec<String>>,
}

/// LLM-Config-Manager adapter for consuming configuration data
pub struct ConfigManagerAdapter {
    config: ConfigManagerConfig,
    connected: AtomicBool,
}

impl ConfigManagerAdapter {
    pub fn new(config: ConfigManagerConfig) -> Self {
        Self {
            config,
            connected: AtomicBool::new(false),
        }
    }

    /// Fetch analytics parameters
    #[instrument(skip(self))]
    pub async fn fetch_analytics_parameters(&self) -> Result<AnalyticsParameters> {
        if !self.connected.load(Ordering::Relaxed) {
            anyhow::bail!("Config-Manager adapter not connected");
        }

        debug!("Fetching analytics parameters from Config-Manager");

        // Placeholder implementation with sensible defaults
        Ok(AnalyticsParameters {
            config_id: uuid::Uuid::new_v4().to_string(),
            version: "1.0.0".to_string(),
            created_at: Utc::now(),
            aggregation: AggregationConfig {
                default_window_minutes: 5,
                rollup_windows: vec![
                    RollupWindow {
                        name: "1min".to_string(),
                        duration_minutes: 1,
                        aggregations: vec!["avg".to_string(), "count".to_string()],
                    },
                    RollupWindow {
                        name: "5min".to_string(),
                        duration_minutes: 5,
                        aggregations: vec!["avg".to_string(), "min".to_string(), "max".to_string(), "count".to_string()],
                    },
                    RollupWindow {
                        name: "1hour".to_string(),
                        duration_minutes: 60,
                        aggregations: vec!["avg".to_string(), "min".to_string(), "max".to_string(), "p50".to_string(), "p95".to_string(), "p99".to_string(), "count".to_string()],
                    },
                ],
                default_percentiles: vec![0.5, 0.9, 0.95, 0.99],
                max_cardinality: 10000,
                enable_histograms: true,
            },
            anomaly_detection: AnomalyDetectionConfig {
                enabled: true,
                algorithm: AnomalyAlgorithm::ZScore,
                sensitivity: 3.0,
                min_data_points: 30,
                evaluation_window_minutes: 15,
                cooldown_minutes: 60,
            },
            forecasting: ForecastingConfig {
                enabled: true,
                model: ForecastModel::ExponentialSmoothing,
                horizon_hours: 24,
                training_window_days: 7,
                update_frequency_hours: 1,
                confidence_level: 0.95,
            },
            alerting: AlertingConfig {
                enabled: true,
                default_severity: AlertSeverity::Warning,
                channels: Vec::new(),
                rate_limit_per_hour: 100,
                grouping_window_minutes: 5,
            },
            sampling: SamplingConfig {
                enabled: false,
                default_rate: 1.0,
                high_volume_rate: 0.1,
                high_volume_threshold_rps: 10000,
                preserve_errors: true,
            },
        })
    }

    /// Fetch retention settings
    #[instrument(skip(self))]
    pub async fn fetch_retention_settings(&self) -> Result<RetentionSettings> {
        if !self.connected.load(Ordering::Relaxed) {
            anyhow::bail!("Config-Manager adapter not connected");
        }

        debug!("Fetching retention settings from Config-Manager");

        // Placeholder implementation with sensible defaults
        Ok(RetentionSettings {
            config_id: uuid::Uuid::new_v4().to_string(),
            version: "1.0.0".to_string(),
            created_at: Utc::now(),
            policies: vec![
                RetentionPolicy {
                    policy_id: "raw-events".to_string(),
                    name: "Raw Events".to_string(),
                    data_type: DataType::RawEvents,
                    retention_days: 7,
                    tier: StorageTier::Hot,
                    compress_after_days: Some(1),
                    archive_after_days: Some(7),
                },
                RetentionPolicy {
                    policy_id: "aggregated-metrics".to_string(),
                    name: "Aggregated Metrics".to_string(),
                    data_type: DataType::AggregatedMetrics,
                    retention_days: 90,
                    tier: StorageTier::Warm,
                    compress_after_days: Some(7),
                    archive_after_days: Some(30),
                },
                RetentionPolicy {
                    policy_id: "traces".to_string(),
                    name: "Traces".to_string(),
                    data_type: DataType::Traces,
                    retention_days: 14,
                    tier: StorageTier::Hot,
                    compress_after_days: Some(3),
                    archive_after_days: Some(14),
                },
            ],
            archival: ArchivalConfig {
                enabled: false,
                destination: ArchivalDestination::S3 {
                    bucket: "analytics-archive".to_string(),
                    prefix: "data/".to_string(),
                },
                compression: CompressionType::Zstd,
                encryption_enabled: true,
            },
            compaction: CompactionConfig {
                enabled: true,
                schedule_cron: "0 2 * * *".to_string(),
                target_file_size_mb: 256,
                max_concurrent_jobs: 4,
            },
        })
    }

    /// Fetch feature flags
    #[instrument(skip(self))]
    pub async fn fetch_feature_flags(&self) -> Result<FeatureFlags> {
        if !self.connected.load(Ordering::Relaxed) {
            anyhow::bail!("Config-Manager adapter not connected");
        }

        debug!("Fetching feature flags from Config-Manager");

        // Placeholder implementation
        Ok(FeatureFlags {
            config_id: uuid::Uuid::new_v4().to_string(),
            flags: HashMap::new(),
            last_updated: Utc::now(),
        })
    }

    /// Fetch environment-specific configuration
    #[instrument(skip(self))]
    pub async fn fetch_environment_config(&self, environment: &str) -> Result<EnvironmentConfig> {
        if !self.connected.load(Ordering::Relaxed) {
            anyhow::bail!("Config-Manager adapter not connected");
        }

        debug!(environment = %environment, "Fetching environment config from Config-Manager");

        // Placeholder implementation
        Ok(EnvironmentConfig {
            environment: environment.to_string(),
            endpoints: HashMap::new(),
            limits: ResourceLimits {
                max_concurrent_queries: 100,
                max_query_timeout_secs: 300,
                max_result_rows: 100000,
                max_memory_mb: 4096,
            },
            security: SecurityConfig {
                require_auth: true,
                allowed_origins: vec!["*".to_string()],
                rate_limit_rps: 1000,
                ip_whitelist: None,
            },
        })
    }

    /// Get a specific configuration value by key
    #[instrument(skip(self))]
    pub async fn get_config_value<T: serde::de::DeserializeOwned>(
        &self,
        key: &str,
    ) -> Result<Option<T>> {
        if !self.connected.load(Ordering::Relaxed) {
            anyhow::bail!("Config-Manager adapter not connected");
        }

        debug!(key = %key, "Fetching config value from Config-Manager");

        // Placeholder implementation
        Ok(None)
    }
}

#[async_trait]
impl EcosystemAdapter for ConfigManagerAdapter {
    #[instrument(skip(self))]
    async fn connect(&self) -> Result<()> {
        info!(endpoint = %self.config.endpoint, "Connecting to LLM-Config-Manager");

        self.connected.store(true, Ordering::Relaxed);

        info!("Successfully connected to LLM-Config-Manager");
        Ok(())
    }

    async fn health_check(&self) -> Result<AdapterHealth> {
        let start = Instant::now();

        if !self.connected.load(Ordering::Relaxed) {
            return Ok(AdapterHealth::unhealthy("config_manager", "Not connected"));
        }

        let latency_ms = start.elapsed().as_millis() as u64;
        Ok(AdapterHealth::healthy("config_manager", latency_ms))
    }

    #[instrument(skip(self))]
    async fn disconnect(&self) -> Result<()> {
        info!("Disconnecting from LLM-Config-Manager");
        self.connected.store(false, Ordering::Relaxed);
        Ok(())
    }
}
